background/web3.js

```diff
@@ -60,20 +62,38 @@ export default class Web3 {
   web3Message (msg) {
     const { payload, origin } = msg;
 
-    return fetch(
-      `http://${this.DAPPS}/rpc/`,
-      {
-        method: 'POST',
-        mode: 'cors',
-        headers: new Headers({
-          'Content-Type': 'application/json',
-          'X-Parity-Origin': origin
-        }),
-        body: JSON.stringify(payload),
-        redirect: 'error',
-        referrerPolicy: 'origin'
-      }
-    ).then((response) => response.json());
+    const request = {
+      method: 'POST',
+      mode: 'cors',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        'X-Parity-Origin': origin
+      }),
+      body: JSON.stringify(payload),
+      redirect: 'error',
+      referrerPolicy: 'origin'
+    };
+    return fetch(`http://${this.DAPPS}/rpc/`, request)
+      .catch(err => {
+        // TODO [ToDr] Get rid of this in the future!
+        // Version 1.7 of Parity runs dapps server on the same port as RPC.
+        // Previous versions were running on :8080.
+        // For backward compatibility we support both cases.
+        const defaultPort = ':' + DEFAULT_CONFIG.DAPPS.split(':')[1];
```

Comment:
Ah, no sorry. It's a `defaultPort` so we need to take it from `DEFAULT_CONFIG`, few lines later we update current port using `this.DAPPS`